Sources/SwiftProtobuf/BinaryDecoder.swift

```diff
@@ -46,12 +50,47 @@ internal struct BinaryDecoder: Decoder {
 
     private var complete: Bool {return available == 0}
 
-    internal init(forReadingFrom pointer: UnsafePointer<UInt8>, count: Int, extensions: ExtensionMap? = nil) {
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      options: BinaryDecodingOptions,
+      extensions: ExtensionMap? = nil
+    ) {
         // Assuming baseAddress is not nil.
         p = pointer
         available = count
         fieldStartP = p
         self.extensions = extensions
+        self.options = options
+        recursionBudget = options.messageDepthLimit
+    }
+
+    internal init(
+      forReadingFrom pointer: UnsafePointer<UInt8>,
+      count: Int,
+      parent: BinaryDecoder
```

Comment:
JSONDecoder does the same thing.

I'm trying to check, but running `Performance/perf_runner.sh -c 40211b54097692154f0636005cddfe9f2c09dfad` seems to be changing files in my local repo, so I'm not sure if I'm running it correctly or not.